src/test/ui/deriving/deriving-all-codegen.rs

```diff
@@ -45,6 +50,14 @@ struct StructGeneric<T, U> {
     u: U,
 }
 
+// A packed, generic tuple struct. Because it is packed, a `T: Copy` bound is
+// added to all impls except for `Default`. This is because we must access
+// fields using copies (e.g. `&{self.0}`), instead of using direct references
+// (e.g. `&self.0`) which may be undefined in a packed struct.
+#[derive(Clone, Copy, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(packed)]
+struct PackedGeneric<T>(T, u32);
```

Comment:
This also derives `Copy`, so shouldn't this get the simple `clone` that just does `*self`?

Reply:
Ah no, it is generic. Please also add a test for a non-generic `derive(Copy, Clone)` on a packed struct.

Reply:
Done.